lib/tests/streamlit/web/server/routes_test.py

```diff
@@ -182,10 +182,31 @@ def get_app(self):
             ]
         )
 
+    @patch_config_options({"global.developmentMode": False})
     def test_allowed_message_origins(self):
         response = self.fetch("/_stcore/allowed-message-origins")
+        response_body = json.loads(response.body)
         self.assertEqual(200, response.code)
         self.assertEqual(
             {"allowedOrigins": ALLOWED_MESSAGE_ORIGINS, "useExternalAuthToken": False},
-            json.loads(response.body),
+            response_body,
+        )
+        # Check that localhost NOT appended/allowed outside dev mode
+        local_host_appended = response_body["allowedOrigins"][-1] == "http://localhost"
+        self.assertEqual(
+            local_host_appended,
+            False,
+        )
+
+    @patch_config_options({"global.developmentMode": True})
+    def test_allowed_message_origins_dev_mode(self):
+        response = self.fetch("/_stcore/allowed-message-origins")
+        self.assertEqual(200, response.code)
+        # Check that localhost has been appended/allowed in dev mode
+        local_host_appended = (
+            json.loads(response.body)["allowedOrigins"][-1] == "http://localhost"
+        )
+        self.assertEqual(
+            local_host_appended,
+            True,
         )
```

Comment:
nit: for both of these, it'd be slightly more correct to test that `http://localhost` does/doesn't appear anywhere in the `allowedOrigins` list since testing whether it's the last element or not relies on an implementation detail